bolts/callbacks/posthoc_eval.py

```diff
@@ -0,0 +1,75 @@
+from collections import Sequence
+
+from kit import gcopy
+import pytorch_lightning as pl
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+import torch
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+
+class PostHocEval(pl.Callback):
+    """
+    Trains and evaluates an auxillary classifier model.
+    Your model should have:
+        - ``self.classifier``
+        - ``self.datamodule``
+        - ``self.clf_epochs``
+        - ``self.trainer''
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.optimizer: Optimizer
```

Comment:
This is (rightfully) unused

Reply:
yeah... should the classifier be an nn.Module, or a pl.LightningModule? I guess the latter - makes life easier all round

Reply:
Keep it as it is for now. While I don't much like the approach because of the assumptions it makes, I don't have in mind a good alternative at the moment.

Reply:
yeah... Looking in the bolts repo they follow this setup of just palming off the responsibility. Maybe this is the best way 🤷‍♂️